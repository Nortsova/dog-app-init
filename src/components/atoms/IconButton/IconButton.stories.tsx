import { IconButton } from '@mui/material';
import type { Meta, StoryObj } from '@storybook/react';

const meta = {
  title: 'Atoms/IconButton',
  component: IconButton,
  argTypes: {
    color: {
      options: ['primary', 'secondary'],
      control: 'radio'
    }
  }
} satisfies Meta<typeof IconButton>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    color: 'primary',
    children: 'icon'
  }
};

export const Secondary: Story = {
  args: {
    color: 'secondary',
    children: 'icon'
  }
};
