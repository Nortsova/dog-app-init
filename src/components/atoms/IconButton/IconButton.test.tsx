import { composeStories } from '@storybook/react';
import * as stories from './IconButton.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('IconButton', () => {
  const { Primary, Secondary } = composeStories(stories);
  testSnapshot(<Primary />);
  testSnapshot(<Secondary />);
});
