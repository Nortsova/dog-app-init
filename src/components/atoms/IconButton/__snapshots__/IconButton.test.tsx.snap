// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should match snapshot Component 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 2.142857142857143rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #921FED;
  border: #921FED;
  box-shadow: none;
  padding: 8px 11.2px;
  border-radius: 30px;
  background-color: #921FED;
  border: #921FED;
  box-shadow: 2px 2px 0px 0px #000000;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-0:hover {
  background-color: none;
  border: none;
}

.emotion-0:active {
  background-color: none;
  border: none;
}

.emotion-0:hover {
  background-color: #A239F4;
  border: #A239F4;
}

.emotion-0:active {
  background-color: #B568F2;
  border: #B568F2;
}

<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium emotion-0"
    tabindex="0"
    type="button"
  >
    icon
  </button>
</div>
`;

exports[`IconButton should match snapshot Component 2`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 2.142857142857143rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #FCFAFA;
  border: #921FED;
  box-shadow: none;
  padding: 6.4px 28px;
  border-radius: 6px;
  background-color: #E6E6E6;
  box-shadow: 3px 3px 0px 0px #000000;
  border: 0.6px solid #090808;
  -webkit-transition: box-shadow .2s ease;
  transition: box-shadow .2s ease;
}

.emotion-0::-moz-focus-inner {
  border-style: none;
}

.emotion-0.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-0 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-0.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-0:hover {
  background-color: none;
  border: none;
}

.emotion-0:active {
  background-color: none;
  border: none;
}

.emotion-0:hover {
  box-shadow: 3px 3px 0px 0px #FFCF32;
  border: 1px solid #FFCF32;
  background-color: transparent;
  outline: none;
}

.emotion-0:active,
.emotion-0:focus {
  background-color: #FFCF32;
  box-shadow: 3px 3px 0px 0px #000000;
  border: 1px solid #000000;
}

<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeMedium emotion-0"
    tabindex="0"
    type="button"
  >
    icon
  </button>
</div>
`;
